Ignore duplicated `ReasoningContentModel` of `MessageModel` in DynamoDB

Yukinobu-Mine opened the pull request:
*Issue #, if available:*
Closes #928 

*Description of changes:*

- Backend
  - Add model validator for `MessageModel`.
    - Ignore `ReasoningContentModel`s in `content` that has the same one in `thinking_log`.
  - Remove code in `usecases/chat.py` that generates duplicate `ReasoningContentModel`.
- Frontend
  - Add a functionality to `AgentToolList` that displays `ReasoningContentModel` in `thinkingLog`.
    - Display the reasoning of the final assistant message after the thinking log.
  - Add `reasoning` state variable and `reasoning` event to `agentThinkingState`.
  - Add `clear` event to `reasoningState`.
  - Send `reasoning` event to `agentThinkingState` when received `AGENT_THINKING` and reasoning present.
  - Merge both XState `agentThink.ts` and `reasoningState.ts` into single state machine

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Yukinobu-Mine committed:
Ignore duplicated `ReasoningContentModel` in DynamoDB
- Backend
  - Add model validator for `MessageModel`.
    - Ignore `ReasoningContentModel`s in `content` that has the same one in `thinking_log`.
  - Remove code in `usecases/chat.py` that generates duplicate `ReasoningContentModel`.
- Frontend
  - Add a functionality to `AgentToolList` that displays `ReasoningContentModel` in `thinkingLog`.
  - Add `reasoning` state variable and `reasoning` event to `agentThinkingState`.
  - Add `clear` event to `reasoningState`.
  - Send `reasoning` event to `agentThinkingState` when received `AGENT_THINKING` and reasoning present.

Yukinobu-Mine committed:
Display the reasoning of the final assistant message after the thinking log.

statefb commented:
Example for the case which has both reasonings for Thinking Log and Message itself. 

<img width="1612" height="568" alt="image" src="https://github.com/user-attachments/assets/8809314a-6fa5-4977-b8c1-1d743a567506" />

Yukinobu-Mine committed:
Unify the state machine used for message streaming.
- Rename `agentThinkState` to `streamingStateMachine`
- Integrate the functionality of `reasoningState` into `streamingStateMachine`.
- Integrate the content manipulation functionality of `usePostMessageStreaming()` into `streamingStateMachine`.

Yukinobu-Mine committed:
Simplify the callback mechanism of `usePostMessageStreaming`.
- Rename `streamingDispatch` to `handleStreamingEvent`.
- In `useChat`, receive message text update notifications directly from `streamingStateMachine`, rather than from `usePostMessageStreaming`.

statefb reviewed (approved):
LGTM!
